umbraco/Umbraco-CMS
# Support blocks in backoffice clipboard

Fixes https://github.com/umbraco/Umbraco-CMS/issues/8990
Fixes https://github.com/umbraco/Umbraco-CMS/issues/8905

**Description:**

Enables Clipboard to contain entries of the type Block. A block contains raw data of its layout, content, and settings.

The feature has been implemented for Block List Property Editor, meaning that copying items of this Property Editor will not include the Settings as well.
Accepting clipboard entries of Blocks has NOT been implemented in Nested Content, meaning that Nested Content cannot receive items from Block List.
Block List does still support Clipboard entries of ElementType, meaning that you can still copy from Nested Content to Block List. (fairly useful for a soft upgrade)

This update also fixed the Clipboard previews, if a Blocks ContentTypeAlias is matched with the clipboard entry alias we will show the Block Catalog appearance on it. Meaning that colors and thumbnails will be present.

**Test Notes:**

- Test copying single entries
- Test copying the whole property
- Ensure that we get Settings when pasting..
- Make sure to test that inner Block Editors are handled, and receives new UDIs.
- Check that copying ElementType from NC to BL still works.
- Check that Block Catalogue thumbnail and colors are shown on clipboard entries.

![image](https://user-images.githubusercontent.com/6791648/96970956-b7ac8c80-1514-11eb-92eb-2ef10850edf5.png)
